### PR TITLE
Added Tab Navigation In the variables dialog

### PR DESCRIPTION
**Pull request by PawBud**
This is a fix regarding the issue: _#1561_

The reason I did not use shouldComponentUpdate is because 

The picture is self-explanatory.

### CODE REVIEW

**Comment by PawBud** [comment]
https://drive.google.com/file/d/1TLm_QTNbpaCjFz5IoJ1lgyG-EE6pji9A/view?usp=sharing
Here is a screen cast video demonstrating how it works after the change

File: newIDE/app/src/VariablesList/index.js [hunk 1]
```diff
@@ -244,7 +244,7 @@ export default class VariablesList extends React.Component<Props, State> {
             success = parentVariable.renameChild(name, text);
           }
 
-          this.setState({
+          React.PureComponent({
```
Comment (4ian) [inline newIDE/app/src/VariablesList/index.js:247]: Unfortunately this does not make sense.
I would recommend spending more time getting familiar with React, what is a state, what is a prop, how a component works.

Comment (PawBud) [reply newIDE/app/src/VariablesList/index.js:247]: I don't mind doing that, but why is the tab working then?

Comment (PawBud) [reply newIDE/app/src/VariablesList/index.js:247]: For now, I think I will close this PR and try to attain a better solution :)

Comment (4ian) [reply newIDE/app/src/VariablesList/index.js:247]: It's not really "working", you removed a setState so there was no-rendering of the components, and also no error shown if you enter a bad name :)

Comment (PawBud) [reply newIDE/app/src/VariablesList/index.js:247]: oh, I think you are right, I will do a course on React and I am sure that  I will get much better at it in a week's time. 
Thanks for the advice.

**Pull request closed without merging**